cpp/src/barretenberg/honk/composer/composer_helper/turbo_plonk_composer_helper.cpp

```diff
@@ -0,0 +1,187 @@
+    circuit_proving_key->recursive_proof_public_input_indices =
+        std::vector<uint32_t>(recursive_proof_public_input_indices.begin(), recursive_proof_public_input_indices.end());
+    circuit_proving_key->contains_recursive_proof = contains_recursive_proof;
+    return circuit_proving_key;
+}
+
+/**
+ * Compute verification key consisting of selector precommitments.
+ *
+ * @return Pointer to created circuit verification key.
+ * */
+template <typename CircuitConstructor>
+std::shared_ptr<bonk::verification_key> TurboPlonkComposerHelper<CircuitConstructor>::compute_verification_key(
+    const CircuitConstructor& circuit_constructor)
+{
+    if (circuit_verification_key) {
+        return circuit_verification_key;
+    }
+    if (!circuit_proving_key) {
+        compute_proving_key(circuit_constructor);
+    }
+
+    circuit_verification_key = compute_verification_key_common(circuit_proving_key, crs_factory_->get_verifier_crs());
+    circuit_verification_key->composer_type = circuit_proving_key->composer_type;
+    circuit_verification_key->recursive_proof_public_input_indices =
+        std::vector<uint32_t>(recursive_proof_public_input_indices.begin(), recursive_proof_public_input_indices.end());
+    circuit_verification_key->contains_recursive_proof = contains_recursive_proof;
+
+    return circuit_verification_key;
+}
+/**
+ * Compute witness polynomials (w_1, w_2, w_3, w_4).
+ *
+ * @details Fills 3 or 4 witness polynomials w_1, w_2, w_3, w_4 with the values of in-circuit variables. The beginning
+ * of w_1, w_2 polynomials is filled with public_input values.
+ * @return Witness with computed witness polynomials.
+ *
+ * @tparam Program settings needed to establish if w_4 is being used.
+ * */
+template <typename CircuitConstructor>
+void TurboPlonkComposerHelper<CircuitConstructor>::compute_witness(const CircuitConstructor& circuit_constructor,
+                                                                   const size_t minimum_circuit_size)
+{
+
+    if (computed_witness) {
+        return;
+    }
+    auto wire_polynomial_evaluations =
+        compute_witness_base(circuit_constructor, minimum_circuit_size, NUM_RANDOMIZED_GATES);
+
+    for (size_t j = 0; j < program_width; ++j) {
+        std::string index = std::to_string(j + 1);
+        circuit_proving_key->polynomial_store.put("w_" + index + "_lagrange",
+                                                  std::move(wire_polynomial_evaluations[j]));
+    }
+    computed_witness = true;
+}
+/**
+ * Create prover.
+ *  1. Compute the starting polynomials (q_l, etc, sigma, witness polynomials).
+ *  2. Initialize StandardProver with them.
+ *  3. Add Permutation and arithmetic widgets to the prover.
+ *  4. Add KateCommitmentScheme to the prover.
+ *
+ * @return Initialized prover.
+ * */
+template <typename CircuitConstructor>
+plonk::TurboProver TurboPlonkComposerHelper<CircuitConstructor>::create_prover(
```

Comment:
I know this function is just carried over from what already existed but it has always bothered me. It doesn't make any sense to do this construct then move when `emplace_back` is already intended to avoid copying by taking the constructor arguments directly. Anyway, not your fault and probably not the time/place for such cleanup.

Reply:
Left as is for now but changed type to `auto`